src/txreconciliation.h

```diff
@@ -0,0 +1,134 @@
+     * Returns false if a peer seems to violate the protocol rules.
+     */
+    bool EnableReconciliationSupport(const NodeId peer_id, bool inbound,
+        bool recon_requestor, bool recon_responder, uint32_t recon_version, uint64_t remote_salt,
+        size_t outbound_flooders);
+
+    /**
+     * Check if peer is registered to perform reconciliations with.
+     */
+    bool IsPeerRegistered(const NodeId peer_id) const;
+
+    /**
+     * Per BIP-330, we may want to flood certain transactions to a subset of peers with whom we
+     * reconcile.
+     * If the peer was not previously registered for reconciliations, returns nullopt.
+     */
+    std::optional<bool> IsPeerChosenForFlooding(const NodeId peer_id) const;
+
+    /**
+     * Removes any reconciliation-related state/notion of the peer.
+     */
+    void RemovePeer(const NodeId peer_id);
+
+    /**
+     * Returns whether a given peer might respond to our reconciliation requests.
+     * If the peer was not previously registered for reconciliations, returns nullopt.
+     */
+    std::optional<bool> IsPeerResponder(const NodeId peer_id) const;
+
+    /**
+     * Returns the size of the reconciliation set we have locally for the given peer.
+     * If the peer was not previously registered for reconciliations, returns nullopt.
+     */
+    std::optional<size_t> GetPeerSetSize(const NodeId peer_id) const;
+
+    /**
+     * Adds new transactions we want to announce to the peer to the local reconciliation set of the
+     * peer, so that those transactions will be reconciled later.
+     */
+    void StoreTxsToAnnounce(const NodeId peer_id, const std::vector<uint256>& txs_to_reconcile);
+
+    /**
+     * If a it's time to request a reconciliation from the peer, this function will return the
+     * details of our local state (local reconciliation set size and local q-coef value), which
+     * should be communicated to the peer so that they better know how to construct a sketch for us.
+     * If the peer was not previously registered for reconciliations, returns nullopt.
+     */
+    std::optional<std::pair<uint16_t, uint16_t>> MaybeRequestReconciliation(const NodeId peer_id);
+
+    /**
+     * Record an (expected) reconciliation request with parameters to respond when time comes. All
+     * initial reconciliation responses will be done at the same time to prevent privacy leaks.
+     */
+    void HandleReconciliationRequest(const NodeId peer_id, uint16_t peer_recon_set_size, uint16_t peer_q);
+
+    /**
+     * Once it's time to respond to reconciliation requests, we construct a sketch from the local
+     * reconciliation set for the peer who requested a reconciliation, and send it to that peer.
+     * If the peer was not previously registered for reconciliations or it's not the time yet,
+     * returns nullopt.
+     */
+    std::optional<std::vector<uint8_t>> MaybeRespondToReconciliationRequest(const NodeId peer_id);
```

Comment:
IMO, it might be better to pass the vector by reference as an argument instead of returning a `std::optional`. I believe same comment can be made for `MaybeRequestReconciliation`/`HandleSketch`/`FinalizeIncomingReconciliation`.

Reply:
And then return bool? Or return void and make vector empty in case of failure? Maybe depending on the usecase.

Reply:
Yes returning a bool. IMHO, boolean return make it easier to read code paths.